k256/src/arithmetic/field.rs

```diff
@@ -272,7 +236,72 @@ impl FieldElement {
 
     #[cfg(test)]
     pub fn modulus_as_biguint() -> BigUint {
-        Self::one().negate(1).to_biguint().unwrap() + 1.to_biguint().unwrap()
+        Self::ONE.negate(1).to_biguint().unwrap() + 1.to_biguint().unwrap()
+    }
+}
+
+impl Field for FieldElement {
+    const ZERO: Self = Self::ZERO;
+    const ONE: Self = Self::ONE;
+
+    fn random(mut rng: impl RngCore) -> Self {
+        let mut bytes = FieldBytes::default();
+
+        loop {
+            rng.fill_bytes(&mut bytes);
+            if let Some(fe) = Self::from_bytes(&bytes).into() {
+                return fe;
+            }
+        }
+    }
+
+    #[must_use]
+    fn square(&self) -> Self {
+        self.square()
+    }
+
+    #[must_use]
+    fn double(&self) -> Self {
+        self.double()
+    }
+
+    fn invert(&self) -> CtOption<Self> {
+        self.invert()
+    }
+
+    fn sqrt(&self) -> CtOption<Self> {
+        self.sqrt()
+    }
+
+    fn sqrt_ratio(num: &Self, div: &Self) -> (Choice, Self) {
+        ff::helpers::sqrt_ratio_generic(num, div)
+    }
+}
+
+impl PrimeField for FieldElement {
+    type Repr = FieldBytes;
+
+    const MODULUS: &'static str =
+        "fffffffffffffffffffffffffffffffffffffffffffffffffffffffefffffc2f";
+    const NUM_BITS: u32 = 256;
+    const CAPACITY: u32 = 255; // TODO(tarcieri): double check this
+    const TWO_INV: Self = Self::ZERO; // TODO
```

Comment:
I calculated these constants a while ago, you can use https://github.com/fjarri/elliptic-curves/commit/d7de9bfb7dbe0e3ddb9fad45506564fb1c8f511f as a reference (or as a test)

Reply:
Aah right, thanks!

Sidebar: in `p256` I was able to constify `FieldElement` enough to calculate them at compile time.

Reply:
Added in [1040324](https://github.com/RustCrypto/elliptic-curves/pull/730/commits/104032401b9411086467e99e4f29fd32588fc83e)